Sources/IntegerUtilities/GCD.swift

```diff
@@ -0,0 +1,49 @@
+//===--- GCD.swift --------------------------------------------*- swift -*-===//
+//
+// This source file is part of the Swift Numerics open source project
+//
+// Copyright (c) 2021 Apple Inc. and the Swift Numerics project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+//
+//===----------------------------------------------------------------------===//
+
+/// The greatest common divisor of `a` and `b`.
+///
+/// If both inputs are zero, the result is zero. If one input is zero, the
+/// result is the absolute value of the other input.
+///
+/// The resulting value must be representable within the value's type. In
+/// particular, the gcd of a signed, fixed-width integer type's minimum
+/// with itself or zero results in a value that cannot be represented:
+///
+///     gcd(Int.min, Int.min)   // Overflow error
```

Comment:
Can we amend this to: "the gcd of a signed, fixed-width integer type's minimum with itself or zero cannot be represented, and results in a trap."

Reply:
I’m fine with that, though I have a slight preference for retaining the colon at the end rather than turning it into a period.

The existing wording was chosen simply to be consistent with [SignedNumeric.negate()](https://github.com/apple/swift/blob/a375ffebcf40e9f3392e297c24afd73c2aeb3623/stdlib/public/core/Integers.swift#L312-L319), except for the colon which I added.

Though reading it again, in the sentence prior to what you quoted, I think the use of “the value” is unclear. For `negate`, “the value” presumably refers to the instance on which the method was called, but `gcd` is not an instance method.

Perhaps it should become either “The resulting value must be representable.” or “The resulting value must be representable within its type.”

Reply:
Okay, I made a change, let me know what you think.

I went with your wording, including the final period instead of colon, except I added parentheses around “or zero”. I can remove them if you prefer.

I also simplified the previous sentence.